aws/resource_aws_appautoscaling_target.go

```diff
@@ -118,6 +116,36 @@ func resourceAwsAppautoscalingTargetRead(d *schema.ResourceData, meta interface{
 	return nil
 }
 
+func resourceAwsAppautoscalingTargetUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appautoscalingconn
+
+	input := &applicationautoscaling.RegisterScalableTargetInput{
+		ResourceId:        aws.String(d.Get("resource_id").(string)),
+		ScalableDimension: aws.String(d.Get("scalable_dimension").(string)),
+		ServiceNamespace:  aws.String(d.Get("service_namespace").(string)),
+	}
+
+	if d.HasChange("max_capacity") {
+		input.MaxCapacity = aws.Int64(int64(d.Get("max_capacity").(int)))
+	}
+
+	if d.HasChange("min_capacity") {
+		input.MinCapacity = aws.Int64(int64(d.Get("min_capacity").(int)))
+	}
+
+	if d.HasChange("role_arn") {
+		input.RoleARN = aws.String(d.Get("role_arn").(string))
```

Comment:
Can you add the role_arn update in tests so that we ensure it's all ok?
Seems rather good for me otherwise :)